src/core_plugins/kibana/public/visualize/listing/visualize_listing.js

```diff
@@ -0,0 +1,167 @@
+  Private,
+  timefilter
+) {
+  timefilter.enabled = false;
+
+  const limitTo = $filter('limitTo');
+  // TODO: Extract this into an external service.
+  const services = Private(SavedObjectRegistryProvider).byLoaderPropertiesName;
+  const visualizationService = services.visualizations;
+  const notify = new Notifier({ location: 'Visualize' });
+
+  let selectedItems = [];
+
+  /**
+   * Sorts hits either ascending or descending
+   * @param  {Array} hits Array of saved finder object hits
+   * @return {Array} Array sorted either ascending or descending
+   */
+  const sortItems = () => {
+    const sortProperty = this.getSortProperty();
+
+    this.items =
+      sortProperty.isAscending
+      ? _.sortBy(this.items, sortProperty.getValue)
+      : _.sortBy(this.items, sortProperty.getValue).reverse();
+  };
+
+  const calculateItemsOnPage = () => {
+    sortItems();
+    this.pager.setTotalItems(this.items.length);
+    this.pageOfItems = limitTo(this.items, this.pager.pageSize, this.pager.startIndex);
+  };
+
+  const fetchObjects = () => {
+    visualizationService.find(this.filter)
+    .then(result => {
+      this.items = result.hits;
+      calculateItemsOnPage();
+    });
+  };
+
+  this.items = [];
+  this.pageOfItems = [];
+  this.filter = '';
+
+  this.pager = pagerService.create(this.items.length, 20, 1);
+
+  /**
+   * Remember sort direction per property.
+   */
+  this.sortProperties = [{
+    name: 'title',
+    getValue: item => item.title,
+    isSelected: true,
+    isAscending: true,
+  }, {
+    name: 'type',
+    getValue: item => item.type.title,
+    isSelected: false,
+    isAscending: true,
+  }];
+
+  this.getSortProperty = function getSortProperty() {
+    return _.find(this.sortProperties, property => property.isSelected);
+  };
+
+  this.getSortPropertyByName = function getSortPropertyByName(name) {
+    return _.find(this.sortProperties, property => property.name === name);
+  };
+
+  this.isAscending = function isAscending() {
+    const sortProperty = this.getSortProperty();
+    return sortProperty.isAscending;
+  };
+
+  this.sortOn = function sortOn(propertyName) {
+    const sortProperty = this.getSortProperty();
+
+    if (sortProperty.name === propertyName) {
+      sortProperty.isAscending = !sortProperty.isAscending;
+    } else {
+      sortProperty.isSelected = false;
+      this.getSortPropertyByName(propertyName).isSelected = true;
+    }
+
+    calculateItemsOnPage();
+  };
+
+  this.toggleAll = function toggleAll() {
+    if (this.areAllItemsChecked()) {
+      selectedItems = [];
+    } else {
+      selectedItems = this.items.slice(0);
+    }
+  };
+
+  this.toggleItem = function toggleItem(item) {
+    if (this.isItemChecked(item)) {
+      const index = selectedItems.indexOf(item);
+      selectedItems.splice(index, 1);
+    } else {
+      selectedItems.push(item);
+    }
+  };
+
+  this.isItemChecked = function isItemChecked(item) {
+    return selectedItems.indexOf(item) !== -1;
```

Comment:
Yep, we use the babel-polyfill so we get all of the new Array and String methods, along with Maps and Sets and such